src/utils.js

```diff
@@ -257,3 +267,265 @@ export function getLinkWithFilename(url) {
 
   return new URL(filename, base);
 }
+
+/**
+ * Helper DocumentationTool class.
+ *
+ * Allows us to detect what doctool is using the current page.
+ *
+ */
+export class DocumentationTool {
+  constructor() {
+    this.documentationTool = this.getDocumentationTool();
+    console.debug(`Documentation tool detected: ${this.documentationTool}`);
+
+    this.DEFAULT_ROOT_SELECTOR = {
+      SPHINX: "[role=main]",
+      MKDOCS_MATERIAL: "main > div > div.md-content",
+      DOCSIFY: "article#main",
+      ASCIIDOCTOR: "div#content",
+      PELICAN: "article",
+      DOCUSAURUS: "article",
+      ANTORA: "article",
+      JEKYLL: "article",
+      FALLBACK_DOCTOOL: ["main", "div.body", "div.document", "body"],
+    };
+
+    this.DEFAULT_LINK_SELECTOR = {
+      sphinx: "a.internal",
+      fallback: ["p a"],
+    };
+  }
+
+  /**
+   * Return the CSS selector to get all the links inside the content of the page.
+   *
+   * The selector returned is based on the documentation tool detected.
+   * If no documentation tool was detected, we iterate over the fallback selectors
+   * and return the first selector that finds an element.
+   */
+  getLinkSelector() {
+    if (
+      this.documentationTool &&
+      objectPath.get(this.DEFAULT_LINK_SELECTOR, this.documentationTool, null)
+    ) {
+      return `${this.getRootSelector()} ${
+        this.DEFAULT_LINK_SELECTOR[this.documentationTool]
+      }`;
+    }
+
+    // Fallback to our list of generic selectors and stop in the first we found.
+    let element;
+    let fullSelector;
+    for (const selector of this.DEFAULT_LINK_SELECTOR.fallback) {
```

Comment:
Assuming class static variable and we want to reuse the constants we started using:

```suggestion
    for (const selector of this.constructor.DEFAULT_LINK_SELECTOR[FALLBACK_DOCTOOL]) {
```